### PR TITLE
100% coverage

### LINKED ISSUE #10: Bring coverage to 100 %
Maybe even by deleting the untested parts if they are not used anywhere.

### PR DESCRIPTION
**Pull request by jamadden**
Fixes #10

### CODE REVIEW

File: src/zope/mimetype/source.py [hunk 1]
```diff
@@ -179,16 +179,16 @@ def preferredCharset(self):
             name=self.value.name)
         if charset is None:
             available = [(name, charset)
-                         for (charset, name) in zope.component.getUtilitiesFor(
+                         for (name, charset) in zope.component.getUtilitiesFor(
                              zope.mimetype.interfaces.ICharset)
                          if charset.encoding == self.value.name]
```
Comment (mgedmin) [inline src/zope/mimetype/source.py]: Nice catch!

File: src/zope/mimetype/tests/test_codec.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+#
+##############################################################################
+# -*- coding: utf-8 -*-
+"""
+Tests for codec.py.
+
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
```
Comment (mgedmin) [inline src/zope/mimetype/tests/test_codec.py]: I don't see you printing or dividing anything in this file.

File: src/zope/mimetype/tests/test_source.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        except ImportError: # pragma: no cover
+            raise unittest.SkipTest("Missing browser extra")
+
+        return CodecTerms(source if source is not None else codecSource,
+                          None)
+
+
+    def setUp(self):
+        from zope.configuration import xmlconfig
+        xmlconfig.string("""
```
Comment (mgedmin) [inline src/zope/mimetype/tests/test_source.py]: Shouldn't you call the superclass `setUp()` at the beginning of this one?

File: src/zope/mimetype/widget.py [hunk 4]
```diff
@@ -52,12 +53,10 @@ def renderItemsWithValues(self, values):
         missing = self._toFormValue(self.context.missing_value)
 
         if self._displayItemForMissingValue and not self.context.required:
-            if missing in values:
-                render = self.renderSelectedItem
-            else:
-                render = self.renderItem
+            render = self.renderSelectedItem if missing in values else self.renderItem
```
Comment (mgedmin) [inline src/zope/mimetype/widget.py]: Ah, cheating on line coverage metrics. ;)

File: src/zope/mimetype/tests/test_codec.py (hunk 2 above)
Comment (jamadden) [reply src/zope/mimetype/tests/test_codec.py]: My standard template puts that at the top of new files. I find it avoids issues in the future. And I probably *was* printing during the development of that incredibly involved test 😉 

I find the `absolute_import` to frequently be necessary when importing from namespaced packages if I don't want pylint to complain, for whatever reason.

File: src/zope/mimetype/widget.py (hunk 4 above)
Comment (jamadden) [reply src/zope/mimetype/widget.py]: I [admitted that](
https://github.com/zopefoundation/zope.mimetype/pull/13/commits/a29d66a9e6e0fe8afbc621d73259304c16923bb2) :)

File: src/zope/mimetype/tests/test_source.py (hunk 3 above)
Comment (jamadden) [reply src/zope/mimetype/tests/test_source.py]: Practically speaking, super's `setUp` is `cleanup.cleanup`, which has nothing to do. But I agree it's good form.

**Comment by icemac** [comment]
@jamadden Thank you for fixing the issue I brought up.

**Pull request merged**